thetis/solver2d.py

```diff
@@ -340,69 +340,92 @@ def create_timestepper(self):
             'momentum_source': self.options.momentum_source_2d,
             'volume_source': self.options.volume_source_2d, }
         self.set_time_step()
+        if self.options.timestepper_type == 'CrankEuler' and not self.options.solve_tracer:
+            self.options.timestepper_type = 'CrankNicolson'
         if self.options.timestepper_type == 'SSPRK33':
-            self.timestepper = rungekutta.SSPRK33(self.eq_sw, self.fields.solution_2d,
-                                                  fields, self.dt,
-                                                  bnd_conditions=self.bnd_functions['shallow_water'],
-                                                  solver_parameters=self.options.timestepper_options.solver_parameters)
+            if self.options.solve_tracer:
```

Comment:
This chain of if clauses is too long. There must be a shorter way of doing this. A good rule in programming is to avoid code duplication (always!). At least it seems that transposing the `timestepper_type` and `solve_tracer` checks would make sense as the coupled integrator construction is always the same.